### DISCUSSION TITLE
New Builds Q4 2024

### DISCUSSION THREAD

**Discussion by ianw1974** [ianw1974/zimbra-build-scripts#93] [Announcements]
Builds this time are for ```9.0.0``` and ```10.1.0```.  As far as I can see from running a test build/upgrade on my Rocky 8 server, it seems to be working OK from basic checks.  All services are working:

```
[root@zimbra10 zcs-10.1.0_GA_0424.RHEL8_64.20241001092503]# /etc/init.d/zimbra status
Host zimbra10.local
	amavis                  Running
	antispam                Running
	antivirus               Running
	dnscache                Running
	ldap                    Running
	logger                  Running
	mailbox                 Running
	memcached               Running
	mta                     Running
	opendkim                Running
	proxy                   Running
	service webapp          Running
	snmp                    Running
	spell                   Running
	stats                   Running
	zimbra webapp           Running
	zimbraAdmin webapp      Running
	zimlet webapp           Running
	zmconfigd               Running
```

Upgrade log - please note you will see additional things that appear that were not there before (and from what I believe these are Zimbra NE components anyway so they shouldn't be installed - which you can see from my deployment below):

```
[root@zimbra10 zcs-10.1.0_GA_0424.RHEL8_64.20241001092503]# ./install.sh 

Operations logged to /tmp/install.log.yjqUULtI
Checking for existing installation...
    zimbra-license-tools...NOT FOUND
    zimbra-license-extension...NOT FOUND
    zimbra-network-store...NOT FOUND
    zimbra-ldap...FOUND zimbra-ldap-10.0.8_GA_0324
    zimbra-logger...FOUND zimbra-logger-10.0.8_GA_0324
    zimbra-mta...FOUND zimbra-mta-10.0.8_GA_0324
    zimbra-dnscache...FOUND zimbra-dnscache-10.0.8_GA_0324
    zimbra-snmp...FOUND zimbra-snmp-10.0.8_GA_0324
    zimbra-license-daemon...NOT FOUND
    zimbra-store...FOUND zimbra-store-10.0.8_GA_0324
    zimbra-apache...FOUND zimbra-apache-10.0.8_GA_0324
    zimbra-spell...FOUND zimbra-spell-10.0.8_GA_0324
    zimbra-convertd...NOT FOUND
    zimbra-memcached...FOUND zimbra-memcached-1.6.5-1zimbra8.7b1
    zimbra-proxy...FOUND zimbra-proxy-10.0.8_GA_0324
    zimbra-archiving...NOT FOUND
    zimbra-onlyoffice...NOT FOUND
    zimbra-core...FOUND zimbra-core-10.0.8_GA_0324
ZCS upgrade from 10.0.8 to 10.1.0 will be performed.
Validating ldap configuration
LDAP validation succeeded.  Continuing.


----------------------------------------------------------------------
PLEASE READ THIS AGREEMENT CAREFULLY BEFORE USING THE SOFTWARE.
SYNACOR, INC. ("SYNACOR") WILL ONLY LICENSE THIS SOFTWARE TO YOU IF YOU
FIRST ACCEPT THE TERMS OF THIS AGREEMENT. BY DOWNLOADING OR INSTALLING
THE SOFTWARE, OR USING THE PRODUCT, YOU ARE CONSENTING TO BE BOUND BY
THIS AGREEMENT. IF YOU DO NOT AGREE TO ALL OF THE TERMS OF THIS
AGREEMENT, THEN DO NOT DOWNLOAD, INSTALL OR USE THE PRODUCT.

License Terms for this Zimbra Collaboration Suite Software:
https://www.zimbra.com/license/zimbra-public-eula-2-6.html
----------------------------------------------------------------------



Do you agree with the terms of the software license agreement? [N] y

Checking current number of databases...
Checking the keysize of /opt/zimbra/conf/ca/ca.key ...

Do you want to verify message store database integrity? [Y] n


Use Zimbra's package repository [Y] 


Installing required dependencies
Configuring package repository

Checking for installable packages

Found zimbra-core (local)
Found zimbra-ldap (local)
Found zimbra-logger (local)
Found zimbra-mta (local)
Found zimbra-dnscache (local)
Found zimbra-snmp (local)
Found zimbra-store (local)
Found zimbra-apache (local)
Found zimbra-spell (local)
Found zimbra-memcached (repo)
Found zimbra-proxy (local)


The Zimbra Collaboration Server appears to already be installed.
It can be upgraded with no effect on existing accounts,
or the current installation can be completely removed prior
to installation for a clean install.

Do you wish to upgrade? [Y] 

Scanning for any new or additional packages available for installation
Existing packages will be upgraded
    Upgrading zimbra-core
    Upgrading zimbra-ldap
    Upgrading zimbra-logger
    Upgrading zimbra-mta
    Upgrading zimbra-dnscache
    Upgrading zimbra-snmp
    Upgrading zimbra-store
    Upgrading zimbra-apache
    Upgrading zimbra-spell
    Upgrading zimbra-memcached
    Upgrading zimbra-proxy
Checking required space for zimbra-core
Checking space for zimbra-store
Checking required packages for zimbra-store
zimbra-store package check complete.

Installing:
    zimbra-core
    zimbra-ldap
    zimbra-logger
    zimbra-mta
    zimbra-dnscache
    zimbra-snmp
    zimbra-store
    zimbra-apache
    zimbra-spell
    zimbra-memcached
    zimbra-proxy

The system will be modified.  Continue? [N] y

Beginning Installation - see /tmp/install.log.yjqUULtI for details...

                          zimbra-core-components will be downloaded and installed.
                          zimbra-common-core-jar will be installed.
                         zimbra-common-core-libs will be installed.
                         zimbra-common-mbox-conf will be installed.
                   zimbra-common-mbox-conf-attrs will be installed.
                    zimbra-common-mbox-conf-msgs will be installed.
                  zimbra-common-mbox-conf-rights will be installed.
                           zimbra-common-mbox-db will be installed.
                         zimbra-common-mbox-docs will be installed.
                   zimbra-common-mbox-native-lib will be installed.
                            zimbra-timezone-data will be installed.
                                     zimbra-core will be installed.
                          zimbra-ldap-components will be downloaded and installed.
                                     zimbra-ldap will be installed.
                                   zimbra-logger will be installed.
                           zimbra-mta-components will be downloaded and installed.
                                      zimbra-mta will be installed.
                      zimbra-dnscache-components will be downloaded and installed.
                                 zimbra-dnscache will be installed.
                          zimbra-snmp-components will be downloaded and installed.
                                     zimbra-snmp will be installed.
Last metadata expiration check: 0:00:32 ago on Tue 01 Oct 2024 10:03:39 AM CEST.
                       zimbra-jetty-distribution will be downloaded and installed.
                         zimbra-store-components will be downloaded and installed.
                   zimbra-mbox-admin-console-war will be installed.
                                zimbra-mbox-conf will be installed.
                                 zimbra-mbox-war will be installed.
                             zimbra-mbox-service will be installed.
                          zimbra-mbox-store-libs will be installed.
                       zimbra-mbox-webclient-war will be installed.
                                    zimbra-store will be installed.
                        zimbra-apache-components will be downloaded and installed.
                                   zimbra-apache will be installed.
                         zimbra-spell-components will be downloaded and installed.
                                    zimbra-spell will be installed.
Last metadata expiration check: 0:00:36 ago on Tue 01 Oct 2024 10:03:39 AM CEST.
                                zimbra-memcached will be downloaded and installed.
                         zimbra-proxy-components will be downloaded and installed.
                                    zimbra-proxy will be installed.

Downloading packages (11):
   zimbra-core-components
   zimbra-ldap-components
   zimbra-mta-components
   zimbra-dnscache-components
   zimbra-snmp-components
   zimbra-jetty-distribution
   zimbra-store-components
   zimbra-apache-components
   zimbra-spell-components
   zimbra-memcached
   zimbra-proxy-components
      ...done

Saving existing configuration file to /opt/zimbra/.saveconfig

Shutting down zimbra mail

Backing up the ldap database...done.
Upgrading the remote packages

Remove zimbra-connect if it is installed ...

Remove zimbra-connect-modern if it is installed ...

Remove zimbra-drive if it is installed ...

Remove zimbra-drive-ng if it is installed ...

Remove zimbra-drive-modern if it is installed ...

Remove zimbra-docs if it is installed ...

Remove zimbra-docs-modern if it is installed ...

Remove zimbra-chat if it is installed ...

Remove zimbra-talk if it is installed ...

Remove zimbra-zimlet-auth if it is installed ...

Remove zimbra-zimlet-briefcase-edit-lool if it is installed ...

Remove zimbra-network-modules-ng if it is installed ...

Remove zimbra-zimlet-restore-contacts if it is installed ...

Remove zimbra-zimlet-duplicate-contacts if it is installed ...

Removing deployed webapp directories

Installing repo packages (11):
   zimbra-core-components
   zimbra-ldap-components
   zimbra-mta-components
   zimbra-dnscache-components
   zimbra-snmp-components
   zimbra-jetty-distribution
   zimbra-store-components
   zimbra-apache-components
   zimbra-spell-components
   zimbra-memcached
   zimbra-proxy-components
      ...done

Installing local packages (26):
   zimbra-common-core-jar
   zimbra-common-core-libs
   zimbra-common-mbox-conf
   zimbra-common-mbox-conf-attrs
   zimbra-common-mbox-conf-msgs
   zimbra-common-mbox-conf-rights
   zimbra-common-mbox-db
   zimbra-common-mbox-docs
   zimbra-common-mbox-native-lib
   zimbra-timezone-data
   zimbra-core
   zimbra-ldap
   zimbra-logger
   zimbra-mta
   zimbra-dnscache
   zimbra-snmp
   zimbra-mbox-admin-console-war
   zimbra-mbox-conf
   zimbra-mbox-war
   zimbra-mbox-service
   zimbra-mbox-store-libs
   zimbra-mbox-webclient-war
   zimbra-store
   zimbra-apache
   zimbra-spell
   zimbra-proxy
      ...done

Running Post Installation Configuration:

Setting defaults from saved config in /opt/zimbra/.saveconfig/config.save
   HOSTNAME=zimbra10.local
   LDAPHOST=zimbra10.local
   LDAPPORT=389
   SNMPTRAPHOST=zimbra10.local
   SMTPSOURCE=admin@zimbra10.local
   SMTPDEST=admin@zimbra10.local
   SNMPNOTIFY=yes
   SMTPNOTIFY=yes
   LDAPROOTPW=*
   LDAPZIMBRAPW=*
   LDAPPOSTPW=*
   LDAPREPPW=*
   LDAPAMAVISPW=*
   LDAPNGINXPW=*
Restoring existing configuration file from /opt/zimbra/.saveconfig/localconfig.xml...done
Copying /tmp/install.log.yjqUULtI to /opt/zimbra/log
Operations logged to /tmp/zmsetup.20241001-100529.log
Adding /opt/zimbra/conf/ca/ca.pem to cacerts
Upgrading from 10.0.8_GA_0324 to 10.1.0_GA_0424
Stopping zimbra services...done.
This appears to be 10.0.8_GA
Starting mysql...done.
Checking ldap status...not running.
Checking ldap status...not running.
Starting ldap...done.
Running mysql_upgrade...done.
Stopping mysql...done.
Updating zimbraLDAPSchemaVersion to version '1717060748'
Updating global config and COS's with attributes introduced after 10.0.8_GA...done.
Stopping ldap...done.
Upgrade complete.

Checking ldap status....not running.
Starting ldap...done.
Setting defaults...done.
Setting defaults from existing config...done.
Checking for port conflicts
Port conflict detected: 53 (zimbra-dnscache)
Port conflicts detected! - Press Enter/Return key to continue 
Setting defaults from ldap...done.
Saving config in /opt/zimbra/config.20083...done.
Operations logged to /tmp/zmsetup.20241001-100529.log
Setting local config values...done.
Initializing core config...Setting up CA...done.
Deploying CA to /opt/zimbra/conf/ca ...done.
Setting replication password...done.
Setting Postfix password...done.
Setting amavis password...done.
Setting nginx password...done.
Creating server entry for zimbra10.local...already exists.
Setting Zimbra IP Mode...done.
Saving CA in ldap...done.
Saving SSL Certificate in ldap...done.
Setting spell check URL...done.
Setting service ports on zimbra10.local...done.
Setting Keyboard Shortcut Preferences...done.
Setting zimbraFeatureTasksEnabled=TRUE...done.
Setting zimbraFeatureBriefcasesEnabled=TRUE...done.
Setting Master DNS IP address(es)...done.
Setting DNS cache tcp lookup preference...done.
Setting DNS cache udp lookup preference...done.
Setting DNS tcp upstream preference...done.
Setting TimeZone Preference...done.
Initializing mta config...done.
Setting services on zimbra10.local...done.
Adding zimbra10.local to zimbraMailHostPool in default COS...done.
Creating user spam.fks_okyes@zimbra10.local...already exists.
Creating user ham.guampaqesl@zimbra10.local...already exists.
Creating user virus-quarantine.o5l3d88r@zimbra10.local...already exists.
Setting spam training and Anti-virus quarantine accounts...done.
Configuring SNMP...done.
Setting up syslog.conf...done.
Setting zimbraReverseProxySSLProtocols...done.
Setting mailboxd_java_options...done.
Setting zimbra_zmjava_options...done.
Setting ldap_common_tlsciphersuite...done.
Setting ldap_common_tlsprotocolmin...done.
Setting amavis_sslversion...done.
Setting java options...done.
Checking ldap status....not running.
Starting ldap...done.
Starting servers...done.
Enabling jetty logging...done.
Checking for deprecated zimlets...done.
Checking for network zimlets in LDAP...done.
Removing network zimlets...
Finished removing network zimlets.
Installing common zimlets...
	com_zimbra_adminversioncheck...done.
	com_zimbra_attachcontacts...done.
	com_zimbra_attachmail...done.
	com_zimbra_bulkprovision...done.
	com_zimbra_cert_manager...done.
	com_zimbra_date...done.
	com_zimbra_email...done.
	com_zimbra_gotourl...done.
	com_zimbra_mailarchive...done.
	com_zimbra_phone...done.
	com_zimbra_proxy_config...done.
	com_zimbra_srchhighlighter...done.
	com_zimbra_tooltip...done.
	com_zimbra_url...done.
	com_zimbra_viewmail...done.
	com_zimbra_webex...done.
	com_zimbra_ymemoticons...done.
Finished installing common zimlets.
Getting list of all zimlets...done.
Updating non-standard zimlets...
Finished updating non-standard zimlets.
Restarting mailboxd...done.
Skipping creation of default domain GAL sync account - existing install detected.

You have the option of notifying Zimbra of your installation.
This helps us to track the uptake of the Zimbra Collaboration Server.
The only information that will be transmitted is:
	The VERSION of zcs installed (10.1.0_GA_0424_RHEL8_64)
	The ADMIN EMAIL ADDRESS created (admin@zimbra10.local)

Notify Zimbra of your installation? [Yes] no
Notification skipped
Setting up zimbra crontab...done.


Moving /tmp/zmsetup.20241001-100529.log to /opt/zimbra/log


Configuration complete - press return to exit 
```

Builds will be uploaded to my site as usual once the builds have finished.  A version for Ubuntu 22.04 will appear in due course once I've specifically made a build manually for it.

**Comment by jeredfloyd** [comment]
It may be worth noting that this upgrade is probably quite critical now; one of the issues patched in 10.0.9 was much more serious than the release notes let on -- it is a full RCE: https://www.theregister.com/2024/10/02/mass_exploitation_of_zimbra_rce/

**Reply by ianw1974** [reply]
This is one thing I don't like about Synacor/Zimbra.  The CVE's are marked, and reserved as shown per their page relating to this and yet they have not released any details whatsoever with a CVE score for people to be aware of what is actually going on.

Zero disclosure on their part, very very poor from a security point of view.

**Comment by ianw1974** [comment]
Thread to discuss the CVE's that Zimbra have patched to be discussed here: https://github.com/ianw1974/zimbra-build-scripts/discussions/96

The OSE/FOSS builds aren't affected, so no requirement for me to rebuild them at this point of time.  Any chat related to the CVE's continue in the linked thread, thx :)

**Comment by ianw1974** [comment]
Added build support for EL9, can be downloaded from my site as usual, or feel free to build yourself if you wish :)

**Reply by jeredfloyd** [reply]
Fantastic! I was going to ask for this (I've been slowly migrating from EL8 to EL9) but figured I'd wait to next quarter since "full support" wasn't until 10.1.2 which was after your Q4 builds.

Is the EL9 build from the same code point as the other Q4 builds, or does it reflect the head of "develop" (hopefully post-10.1.2) at the time of build?

**Reply by ianw1974** [reply]
It's possible Synacor/Zimbra made commits between October 1 and today, so they are unlikely to be from the same code point.  It's possible that they are, but without checking 80+ repositories from Zimbra's side, I can't guarantee it.  Had they been built from tags, then it would be a no-brainer, but from the develop branch, difficult to guarantee.

**Comment by jeredfloyd** [comment]
So far, so good here!  Updated from 10.0.8_GA_0224 to 10.1.1_GA_0424.

On 10.0.8 I began encountering periodic lock failures on mailboxes, which would render an account unusable for hours at a time.  (Like, neither IMAP nor WebUI would even load.)  I suspect this is a zimbra database issue with large/sparse mailboxes, because I've encountered multiple mysql queries with tens-of-minute run times.  I'm going to ask on the Zimbra forums about this, but I don't expect much engagement sadly.  I suspect this is related to something weird Apple has done in a recent iOS update.

The queries look like the below and go on for, literally, tens of thousands of lines:
```SELECT mi.id, mi.index_id, mi.type FROM mboxgroup27.mail_item AS mi WHERE mi.mailbox_id = 27 AND ((((mi.type = 5 OR mi.type = 6 OR mi.type = 16)) AND (((mi.id >= 411 AND mi.id <= 414)) OR ((mi.id >= 416 AND mi.id <= 420)) OR ((mi.id >= 422 AND mi.id <= 430)) OR ((mi.id >= 432 AND mi.id <= 456)) OR ((mi.id >= 460 AND mi.id <= 463)) OR ((mi.id >= 465 AND mi.id <= 471)) OR ((mi.id >= 473 AND mi.id <= 483)) OR ((mi.id >= 485 AND mi.id <= 488)) OR ((mi.id >= 490 AND mi.id <= 491)) OR ((mi.id >= 493 AND mi.id <= 494)) OR ((mi.id >= 496 AND mi.id <= 499)) OR ((mi.id >= 501 AND mi.id <= 508)) OR ((mi.id >= 510 AND mi.id <= 514)) OR ((mi.id >= 520 AND mi.id <= 533)) OR ((mi.id >= 535 AND mi.id <= 539)) OR ((mi.id >= 541 AND mi.id <= 545)) OR ((mi.id >= 547 AND mi.id <= 548)) OR ((mi.id >= 550 AND mi.id <= 556)) OR ((mi.id >= 558 AND mi.id <= 566)) OR ((mi.id >= 568 AND mi.id <= 569)) OR ((mi.id >= 573 AND mi.id <= 576)) OR ((mi.id >= 578 AND mi.id <= 580)) OR ((mi.id >= 584 AND mi.id <= 586)) OR ((mi.id >= 588 AND mi.id <= 592)) OR ((mi.id >= 594 AND mi.id <= 598)) OR ((mi.id >= 600 AND mi.id <= 626)) OR ((mi.id >= 628 AND mi.id <= 635)) OR ((mi.id >= 637 AND mi.id <= 640)) OR ((mi.id >= 642 AND mi.id <= 643)) OR ((mi.id >= 645 AND mi.id <= 650)) OR ((mi.id >= 652 AND mi.id <= 668)) OR ((mi.id >= 670 AND mi.id <= 677)) OR ((mi.id >= 679 AND mi.id <= 683)) OR ((mi.id >= 685 AND mi.id <= 689)) OR ((mi.id >= 691 AND mi.id <= 706)) OR ((mi.id >= 708 AND mi.id <= 720)) OR ((mi.id >= 722 AND mi.id <= 724)) OR ((mi.id >= 726 AND mi.id <= 733)) OR ((mi.id >= 735 AND mi.id <= 743)) OR ((mi.id >= 745 AND mi.id <= 752)) OR ((mi.id >= 754 AND mi.id <= 774)) OR ((mi.id >= 776 AND mi.id <= 797)) OR ((mi.id >= 799 AND mi.id <= 800)) OR ((mi.id >= 802 AND mi.id <= 803)) OR ((mi.id >= 805 AND mi.id <= 809)) OR ((mi.id >= 811 AND mi.id <= 814)) OR ((mi.id >= 816 AND mi.id <= 817)) OR ((mi.id >= 819 AND mi.id <= 821)) OR ((mi.id >= 823 AND mi.id <= 836)) OR ((mi.id >= 838 AND mi.id <= 842)) OR ((mi.id >= 846 AND mi.id <= 857)) OR ((mi.id >= 859 AND mi.id <= 862)) OR ((mi.id >= 866 AND mi.id <= 871)) OR ((mi.id >= 873 AND mi.id <= 876)) OR ((mi.id >= 878 AND mi.id <= 891)) OR ((mi.id >= 893 AND mi.id <= 900)) OR ((mi.id >= 902 AND mi.id <= 904)) OR ((mi.id >= 906 AND mi.id <= 907)) OR ((mi.id >= 909 AND mi.id <= 912)) OR ((mi.id >= 914 AND mi.id <= 917)) OR ((mi.id >= 919 AND mi.id <= 921)) OR ((mi.id >= 923 AND mi.id <= 925)) OR ((mi.id >= 927 AND mi.id <= 930)) OR ((mi.id >= 932 AND mi.id <= 942)) OR ((mi.id >= 944 AND mi.id <= 945)) OR ((mi.id >= 947 AND mi.id <= 950)) OR ((mi.id >= 954 AND mi.id <= 955)) OR ((mi.id >= 957 AND mi.id <= 959)) OR ((mi.id >= 963 AND mi.id <= 966)) OR ((mi.id >= 968 AND mi.id <= 972)) OR ((mi.id >= 974 AND mi.id <= 980)) OR ((mi.id >= 982 AND mi.id <= 983)) OR ((mi.id >= 985 AND mi.id <= 988)) OR ((mi.id >= 990 AND mi.id <= 992)) OR ((mi.id >= 994 AND mi.id <= 997)) OR ((mi.id >= 999 AND mi.id <= 1006)) OR ((mi.id >= 1008 AND mi.id <= 1011)) OR ((mi.id >= 1013 AND mi.id <= 1014)) OR ((mi.id >= 1016 AND mi.id <= 1022)) OR ((mi.id >= 1024 AND mi.id <= 1026)) OR ((mi.id >= 1028 AND mi.id <= 1029)) OR ((mi.id >= 1031 AND mi.id <= 1035)) OR ((mi.id >= 1037 AND mi.id <= 1039)) OR ((mi.id >= 1041 AND mi.id <= 1044)) OR ((mi.id >= 1046 AND mi.id <= 1050)) OR ((mi.id >= 1052 AND mi.id <= 1065)) OR ((mi.id >= 1067 AND mi.id <= 1069)) OR ((mi.id >= 1071 AND mi.id <= 1072)) OR ((mi.id >= 1074 AND mi.id <= 1079)) OR ((mi.id >= 1081 AND mi.id <= 1086)) OR ((mi.id >= 1090 AND mi.id <= 1091)) OR ((mi.id >= 1093 AND mi.id <= 1098)) OR ((mi.id >= 1100 AND mi.id <= 1111)) OR ((mi.id >= 1113 AND mi.id <= 1115)) OR ((mi.id >= 1119 AND mi.id <= 1120)) OR ((mi.id >= 1122 AND mi.id <= 1125)) OR ((mi.id >= 1127 AND mi.id <= 1135)) OR ((mi.id >= 1137 AND mi.id <= 1141)) OR ((mi.id >= 1143 AND mi.id <= 1146)) OR ((mi.id >= 1148 AND mi.id <= 1152)) OR ((mi.id >= 1154 AND mi.id <= 1159)) OR ((mi.id >= 1161 AND mi.id <= 1162)) OR ((mi.id >= 1164 AND mi.id <= 1169)) OR ((mi.id >= 1171 AND mi.id <= 1175)) OR ((mi.id >= 1177 AND mi.id <= 1178)) OR ((mi.id >= 1180 AND mi.id <= 1182)) OR ((mi.id >= 1188 AND mi.id <= 1192)) OR ((mi.id >= 1194 AND mi.id <= 1201)) OR ((mi.id >= 1203 AND mi.id <= 1204)) OR ((mi.id >= 1206 AND mi.id <= 1216)) OR ((mi.id >= 1218 AND mi.id <= 1223)) OR ((mi.id >= 1225 AND mi.id <= 1240)) OR ((mi.id >= 1242 AND mi.id <= 1247)) OR ((mi.id >= 1249 AND mi.id <= 1255)) OR ((mi.id >= 1257 AND mi.id <= 1268)) OR ((mi.id >= 1270 AND mi.id <= 1284)) OR ((mi.id >= 1286 AND mi.id <= 1289)) OR ((mi.id >= 1291 AND mi.id <= 1294)) OR ((mi.id >= 1296 AND mi.id <= 1305)) OR ((mi.id >= 1309 AND mi.id <= 1311)) OR ((mi.id >= 1313 AND mi.id <= 1318)) OR ((mi.id >= 1320 AND mi.id <= 1325)) OR ((mi.id >= 1327 AND mi.id <= 1335)) OR ((mi.id >= 1337 AND mi.id <= 1345)) OR ((mi.id >= 1347 AND mi.id <= 1351)) OR ((mi.id >= 1353 AND mi.id <= 1355)) OR ((mi.id >= 1357 AND mi.id <= 1358)) OR ((mi.id >= 1360 AND mi.id <= 1361)) ...```

**Reply by jeredfloyd** [reply]
I see other people are complaining in the forums about recent "mailbox locked" issues and I'm quite certain that this is an iOS 18-driven problem.  Apple is horrible about acknowledging and fixing issues, so Zimbra will have to figure out a work-around.  I created a new thread if anyone here is interested: https://forums.zimbra.org/viewtopic.php?t=72998

**Reply by jeredfloyd** [reply]
This is fixed (or, at least, there is a new config option) upstream, but not yet merged to develop: https://github.com/Zimbra/zm-mailbox/commit/8b145b31bf02343a8e594bbbe84d5e2c1ea8021b

It appears to be a new config option that eliminates UID subrange requests in IMAP SEARCH, which I suppose is one way to do it.  It is targeted for 10.1.4.
